packages/plugin-core/src/commands/CopyNoteLink.ts

```diff
@@ -29,24 +37,33 @@ export class CopyNoteLinkCommand extends BasicCommand<
     window.showInformationMessage(`${link} copied`);
   }
 
-  async execute(_opts: CommandOpts) {
-    const editor = VSCodeUtils.getActiveTextEditor() as TextEditor;
-    const fname = NoteUtils.uri2Fname(editor.document.uri);
-
-    const vault = PickerUtilsV2.getOrPromptVaultForOpenEditor();
-    const notes = getEngine().notes;
-    const note = NoteUtils.getNoteByFnameV5({
-      fname,
-      vault,
-      notes,
-      wsRoot: getDWorkspace().wsRoot,
-    }) as NoteProps;
-    if (!note) {
-      throw Error(
-        `${fname} not found in engine! Try saving this file and running "Dendron: Reload Index"`
-      );
+  private async createNonNoteFileLink(editor: TextEditor) {
+    const { wsRoot, vaults } = getDWorkspace();
+    let { fsPath } = editor.document.uri;
+    // Find it relative to wsRoot
+    fsPath = path.relative(wsRoot, fsPath);
+    // Check if the file is in the assets of any vault. If it is, we can shorten the link.
+    for (const vault of vaults) {
+      const vaultPath = VaultUtils.getRelPath(vault);
+      if (isInsidePath(vaultPath, fsPath)) {
+        fsPath = path.relative(vaultPath, fsPath);
+        break;
+      }
     }
+    let anchor = "";
+    if (!editor.selection.isEmpty) {
```

Comment:
from an UX point, when we introduce anchors, how will we diffentiate the user wanting to copy a line number vs copy an anchor?

Reply:
I think anchors would be the default option, with a configuration to allow line numbers instead. Since line numbers would get stale too quickly, I think we should avoid encouraging their use. It'll still be an option if a user doesn't want to introduce anchors into their code base.

Reply:
sounds good. in a separate PR, can we add the following option?

```yml
commands:
  copyNoteLink:
    nonNote:
      defaultAnchorType: line|block|none
```

by default, this should be "none" until the user chooses to enable one. when we introduce the block anchor, we can switch the default to "block". I want to avoid introducing the "line" default only to switch it later.